include/mbedtls/x509_crt.h

```diff
@@ -986,15 +987,42 @@ void mbedtls_x509write_crt_init(mbedtls_x509write_cert *ctx);
  */
 void mbedtls_x509write_crt_set_version(mbedtls_x509write_cert *ctx, int version);
 
+#if defined(MBEDTLS_BIGNUM_C) && !defined(MBEDTLS_DEPRECATED_REMOVED)
 /**
  * \brief           Set the serial number for a Certificate.
  *
+ * \deprecated      This function is deprecated and will be removed in a
+ *                  future version of the library. Please use
+ *                  mbedtls_x509write_crt_set_serial_new() instead.
+ *
+ * \note            Even though the MBEDTLS_BIGNUM_C guard looks redundant since
+ *                  X509 depends on PK and PK depends on BIGNUM, this emphasizes
+ *                  a direct dependency between X509 and BIGNUM which is going
+ *                  to be deprecated in the future.
+ *
  * \param ctx       CRT context to use
  * \param serial    serial number to set
  *
  * \return          0 if successful
  */
-int mbedtls_x509write_crt_set_serial(mbedtls_x509write_cert *ctx, const mbedtls_mpi *serial);
+int MBEDTLS_DEPRECATED mbedtls_x509write_crt_set_serial(
+    mbedtls_x509write_cert *ctx, const mbedtls_mpi *serial);
+#endif // MBEDTLS_BIGNUM_C && !MBEDTLS_DEPRECATED_REMOVED
+
+/**
+ * \brief           Set the serial number for a Certificate.
+ *
+ * \param ctx          CRT context to use
+ * \param serial       A raw array of bytes containing the serial number in big
+ *                     endian format
+ * \param serial_len   Length of the previous input buffer buffer
```

Comment:
We don't really care about the size of the previous input buffer ~~buffer~~, we care about the length of passed serial data though, right? `cert_write.c` doesn't pass `MBEDTLS_X509_RFC5280_MAX_SERIAL_LEN` in, but `serial_len`.

Reply:
Yes, probably instead of writing "previous input buffer" I was meaning "provided input buffer" (which I used also below), but I really cannot tell how I managed to write such a non-sense sentence :/

Reply:
What I mean is that we don't care about the length of the input buffer.
We care about the data that's inside, and that's the length we should be passing :)

Reply:
Btw, I just remembered: we should get into the habit of making units explicit (most of the time it will be bytes, but sometimes keys are measured in bits, so let's be explicit everywhere). So, `Length of \p serial in bytes` then?

Reply:
> We care about the data that's inside

Ok, now I get the point! Sorry for taking so long to understand that, but you're right and this should be fixed.